libcontainer/nsenter/nsexec.c

```diff
@@ -399,9 +472,40 @@ void join_namespaces(char *nslist)
 		ns->fd = fd;
 		ns->ns = nsflag(namespace);
 		strncpy(ns->path, path, PATH_MAX);
+
+		/* Cache the user namespace entry. */
+		if (ns->ns == CLONE_NEWUSER || !strcmp(namespace, "user"))
+			userns = ns;
```

Comment:
IIRC I wrote it like this to avoid issues where `CLONE_NEWUSER` is not defined by glibc (so `nsflags` would return 0). Though I might have fixed that in the meantime, since `CLONE_NEWUSER` is required by a bunch of other things.